src/jit/codegencommon.cpp

```diff
@@ -7329,12 +7335,35 @@ void CodeGen::genProfilingEnterCallback(regNumber initReg, bool* pInitRegZeroed)
                 continue;
             }
 
-            var_types storeType = varDsc->lvaArgType();
-            regNumber argReg    = varDsc->lvArgReg;
-            getEmitter()->emitIns_S_R(ins_Store(storeType), emitTypeSize(storeType), argReg, varNum, 0);
+#ifdef FEATURE_UNIX_AMD64_STRUCT_PASSING
+            if (varTypeIsStruct(varDsc))
+            {
+                var_types       type0, type1;
+                unsigned __int8 offset0, offset1;
+                compiler->GetStructTypeOffset(varDsc->lvVerTypeInfo.GetClassHandle(), &type0, &type1, &offset0,
+                                              &offset1);
+                getEmitter()->emitIns_S_R(ins_Store(type0), emitTypeSize(type0), varDsc->lvArgReg, varNum, offset0);
+                if (type1 != TYP_UNKNOWN)
+                {
+                    getEmitter()->emitIns_S_R(ins_Store(type1), emitTypeSize(type1), varDsc->lvOtherArgReg, varNum,
+                                              offset1);
+                }
+            }
+            else
+#endif
+            {
+                var_types storeType = varDsc->lvaArgType();
+                regNumber argReg    = varDsc->lvArgReg;
+                getEmitter()->emitIns_S_R(ins_Store(storeType), emitTypeSize(storeType), argReg, varNum, 0);
+            }
         }
     }
 
+#if defined(UNIX_AMD64_ABI)
+    getEmitter()->emitIns_R_R(INS_mov, emitTypeSize(TYP_ULONG), REG_R12, REG_RDI);
+    getEmitter()->emitIns_R_R(INS_mov, emitTypeSize(TYP_ULONG), REG_R13, REG_RSI);
+#endif
```

Comment:
@sivarv RDI and RSI are used for two Enter callback arguments. But as @janvorli required recently we need the original RDI & RSI values to store them in the struct _PROFILE_PLATFORM_SPECIFIC_DATA